### ISSUE TITLE
Incorrect `to_s` for `Yield` node

### ISSUE THREAD

**Issue by asterite** [crystal-lang/crystal#11601]
Take a look at this program:

```crystal
record OperatorPacket do
  def binary_operator : Int32
    yield(1) ? 1 : 0
  end
end

OperatorPacket.new.binary_operator { 1 < 2 }
```

It gives an error:

```
In 16/16.cr:2:25

 2 | def binary_operator : Int32
                           ^----
Error: method OperatorPacket#binary_operator must return Int32 but it is returning Bool
```

The problem is that the method definition happens inside a `do` block of a macro, so that's all parsed and put into the final program again. 

If we use a `show` macro to reveal what's being pasted:

```crystal
macro show(obj)
  {% puts obj %}
end

show(yield(1) ? 1 : 0)
```

We see this:

```crystal
yield 1 ? 1 : 0
```

The main issue here is that `yield` isn't put inside parentheses, and because `? :` has a greater operator precedence, it's parsed like:

```crystal
yield(1 ? 1 : 0)
```

**Comment by straight-shoota** [comment]
Isn't this the same as #10984? It should already be fixed in master (#11469).

**Comment by asterite** [comment]
Indeed 🎉
